tools/clang/lib/Sema/SemaExpr.cpp

```diff
@@ -9417,6 +9446,11 @@ static void DiagnoseConstAssignment(Sema &S, const Expr *E,
       }
       break;
     } // End MemberExpr
+    else if (const ArraySubscriptExpr *ASE = dyn_cast<ArraySubscriptExpr>(E)) {
+      const Expr *Base = ASE->getBase();
+      if (Base->getType()->isMatrixType())
+        E = Base;
```

Comment:
Shouldn't this `continue;` after assigning `E = Base;`?  Or why is this in the loop in the first place?

Reply:
Moved out of the loop.